core/src/components/modal/gestures/sheet.ts

```diff
@@ -42,6 +43,27 @@ export const createSheetGesture = (
   const backdropAnimation = animation.childAnimations.find(ani => ani.id === 'backdropAnimation');
   const maxBreakpoint = breakpoints[breakpoints.length - 1];
 
+  /* tslint:disable-next-line */
+  if (typeof window !== 'undefined') {
+    const keyboardOpenCallback = () => {
+      /**
+       * When the native keyboard is opened and the webview
+       * is resized, the gesture implementation will become unresponsive
+       * and enter a free-scroll mode.
+       *
+       * When the keyboard is opened, we disable the gesture for
+       * a single frame and re-enable once the contents have repositioned
+       * from the keyboard placement.
+       */
+      gesture.enable(false);
+      raf(() => {
+        gesture.enable(true)
+      });
+    };
+
+    window.addEventListener(KEYBOARD_DID_OPEN, keyboardOpenCallback);
```

Comment:
It looks like this only gets called once when the modal is presented, but do we need to remove the listener when it's dismissed? 🤔

Reply:
That's a good point. I had similar concessions, but wasn't able to find anywhere in gestures that we expose appending additional behavior to teardown. Let me see if there's a way to extend the `destroy` method of the gesture (which is called in the dismiss callback from the modal).